dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/PreCheckSecurityOwnershipValidationHook.java

```diff
@@ -83,6 +84,24 @@ public void validateTrackedEntity( ValidationErrorReporter reporter, TrackedEnti
         User user = bundle.getUser();
 
         checkNotNull( user, USER_CANT_BE_NULL );
+        checkNotNull( trackedEntity, TRACKED_ENTITY_CANT_BE_NULL );
+        checkNotNull( trackedEntity.getOrgUnit(), ORGANISATION_UNIT_CANT_BE_NULL );
+
+        boolean isNew = context.getTrackedEntityInstance( trackedEntity.getTrackedEntity() ) == null;
+
+        // If trackedEntity is newly created, or going to be deleted, capture
+        // scope has to be checked
+        if ( isNew || strategy.isDelete() )
```

Comment:
Could we use the strategy instead of getting the object from the preheat?
The strategy is calculated per object, so the isNew logic already happened in StrategyPreProcessor

```suggestion
        if ( strategy.isCreate() || strategy.isDelete() )
```

Reply:
Very good suggestion. I wasnt aware that strategy reflects per object strategy. Thanks!